src/syntax-highlighting.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  buildRegExp,
+  capture,
+  charClass,
+  choiceOf,
+  endOfString,
+  notWhitespace,
+  startOfString,
+  whitespace,
+  word,
+  zeroOrMore,
+} from 'ts-regex-builder';
+
+const BACKTICKS = '```';
+
+export const codeBlockRegExp = buildRegExp(
+  [capture([BACKTICKS, zeroOrMore(charClass(whitespace, notWhitespace)), BACKTICKS])],
+  {
+    global: true,
+    ignoreCase: true,
+  }
+);
+
+export const codeBlockAnnotationsRegExp = buildRegExp(
+  choiceOf([startOfString, BACKTICKS, zeroOrMore(word)], [BACKTICKS, endOfString]),
+  {
+    global: true,
+    ignoreCase: true,
+  }
+);
+
+export const langRegExp = buildRegExp([BACKTICKS, capture(zeroOrMore(word))], {
+  ignoreCase: true,
+});
+
+export function splitCodeBlocks(text: string): string[] {
+  return text.split(codeBlockRegExp);
+}
+
+export function stripCodeblockAnnotations(text: string): string {
```

Comment:
Looks like we could use one smarter regex to match all of the following:
1. a code block
2. language
3. actual content

Let's try:
```
// Question does language name requires adding new line?
const openingCodeBlock = [BACKTICKS, capture(zeroOrMore(word)), '\n'];

export const codeBlockRegExp = buildRegExp(
  [
    openingCodeBlock,
    capture(oneOrMore(any, { greedy: false })), // actual code
    BACKTICKS,
  ],
  {
    global: true,
    multiline: true, // We should need to add this one, to avoid RegExp treating every line like a separate search string.
    // ignoreCase: true does not seem to be used
  }
```

The following expression should capture language name (potentially empty) in the first capture group, and code block content in the second capture group.